### PR TITLE
Added blue color to tanks

### LINKED ISSUE #5: Add color to the tank

### PR DESCRIPTION
**Pull request by pridhviu**
I want to resolve #5 by adding BLUE colour to the tanks while rendering main playing screen.

### CODE REVIEW

**Comment by pridhviu** [comment]
Added collision detection and boundary checking to resolve #9

**Comment by thuanpv** [comment]
Hi @pridhvi. Thank you for your first PR. Regarding the changes to add colors for the tanks, I expect a more generic solution that can apply to tanks as well as other objects on the playfield. I was thinking of adding a color attribute.

Regarding the collision detection algorithm, it would be great if you could present your solution to your class mates in the live lecture tomorrow.

**Comment by pridhviu** [comment]
Hi @thuanpv I have done the following,

- Removed the blue colour printing in the previous commit.
- Added a new class Object which acts as the parent class for all objects in the game map including Tanks and other obstacles. I have made Tank class to inherit from Object class. This object class has a color attribute which can used when implementing the color render.

File: CityTankGame.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ public static void main(String[] args)
 
       //Get user command
       System.out.println("Control your tank by typing a character and press Enter. ");
-      System.out.println("W: up, A: left, S: right, Z: down, Q: quit the game.");
+      System.out.println("W: up, A: left, D: right, S: down, Q: quit the game.");
```
Comment (essut) [inline CityTankGame.java:37]: I don't think this needs to be changed. It is valid to use a non-WASD control.

File: objects/Object.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package objects;
```
Comment (essut) [inline objects/Object.java:1]: This is implementing the game using inheritance. Can you kindly split the pull requests so it is easier to review?

File: objects/Tank.java [hunk 3]
```diff
@@ -57,10 +31,10 @@ public void move(Commands.Direction cmd)
         switch (cmd)
         {
           case UP:
-            rowPos--;
+            if(! isCollision(mainRenderer, getRowPos() - 1, getColPos())) setRowPos(getRowPos() - 1);
```
Comment (essut) [inline objects/Tank.java:34]: I don't prefer this kind of if statement. Wouldn't it be better if the condition and statement is split into separate lines?

File: objects/Tank.java [hunk 4]
```diff
@@ -171,4 +141,25 @@ public void move(Commands.Direction cmd)
     }
     return bitmap;
   }
-}
+
+  public boolean isCollision(Renderer mainRenderer, int rowPos, int colPos) {
```
Comment (essut) [inline objects/Tank.java:145]: I think this is a helper function, which should be private.

File: CityTankGame.java (hunk 1 above)
Comment (pridhviu) [reply CityTankGame.java]: I have reverted it.

File: objects/Tank.java (hunk 3 above)
Comment (pridhviu) [reply objects/Tank.java]: I have made the if condition into multiple lines

File: objects/Tank.java (hunk 4 above)
Comment (pridhviu) [reply objects/Tank.java]: Thanks for the suggestion I missed that. I have made it private now

File: objects/Object.java (hunk 2 above)
Comment (pridhviu) [reply objects/Object.java]: I have created a separate PR for this. #12

**Comment by thuanpv** [comment]
I close this PR because the newer one has been merged. Thanks you both! Keep up the good work!

**Pull request closed without merging**